src/effects/imagefilters/SkBlurImageFilter.cpp

```diff
@@ -146,14 +146,20 @@ class Pass {
             if (int commonEnd = std::min(dstIdx, srcEnd); srcIdx < commonEnd) {
                 // Preload the blur with values from src before dst is entered.
                 int n = commonEnd - srcIdx;
-                this->blurSegment(n, srcCursor, srcStride, nullptr, 0);
-                srcIdx += n;
+                this->blurSegment(n, srcCursor, srcStride, nullptr, 0, 0);
+                
+                if (fTileMode == SkTileMode::kMirror) {
+                    // This ensures that we don't bleed any 0 pixels into the output at the starting edges
+                    this->blurSegment(n, srcCursor, srcStride, nullptr, 0, 0);
```

Comment:
so this is applied twice with the same parameters?

Reply:
Yes, but I've also fixed it now to actually sample in the correct order (had to go in reverse here)